sdks/python/apache_beam/testing/test_stream.py

```diff
@@ -66,6 +68,28 @@ def __ne__(self, other):
     # TODO(BEAM-5949): Needed for Python 2 compatibility.
     return not self == other
 
+  @abstractmethod
+  def to_runner_api(self, element_coder):
+    raise NotImplementedError
+
+  @staticmethod
+  def from_runner_api(proto, element_coder):
```

Comment:
For my understanding: Is the reason to group the from_runner_api implementation into the abstract class (while triggers/windows leave the from_runner_api to the concrete classes) that we believe TestStream will only have very limited types of element?

Reply:
Yes, exactly. This is not a user-extensible set. (Also, the logic is trivial, so not worth the noise of delegating.)